bundles/org.openhab.voice.voicerss/src/main/java/org/openhab/voice/voicerss/internal/cloudapi/VoiceRSSCloudImpl.java

```diff
@@ -121,15 +175,21 @@
 
     @Override
     public Set<String> getAvailableVoices() {
-        return SUPPORTED_VOICES;
+        // different locales support different voices, so let's list all here in one big set when no locale is provided
+        Set<String> allvoxes = new HashSet<>();
+        for (Set<String> langvoxes : SUPPORTED_VOICES.values()) {
+            for (String langvox : langvoxes) {
+                allvoxes.add(langvox);
+            }
+        }
+        return allvoxes;
     }
 
     @Override
     public Set<String> getAvailableVoices(Locale locale) {
-        for (Locale voiceLocale : SUPPORTED_LOCALES) {
-            if (voiceLocale.toLanguageTag().equalsIgnoreCase(locale.toLanguageTag())) {
-                return SUPPORTED_VOICES;
-            }
+        String langtag = locale.toLanguageTag();
+        if (SUPPORTED_VOICES.containsKey(langtag)) {
```

Comment:
By suppressing the ignore case, that means it will no more work if the locale is fr-FR for example,

Reply:
but is it possible to containskey case insensitive?

Reply:
I assume all the keys in your map SUPPORTED_VOICES are in lowercase (to be verified).
You should then check what returns locale.toLanguageTag()
If the result is not in lowercase, you can simply replace your line by
```
String langtag = locale.toLanguageTag().toLowercase();
```